File: skills/vuln-discovery/evals/fixtures/go/path_traversal_vulnerable.go
// Vulnerable: Path traversal via user-controlled filename in file serve.
package main

import (
	"net/http"
	"os"
	"path/filepath"
)

const baseDir = "/var/data/files"

func downloadHandler(w http.ResponseWriter, r *http.Request) {
	filename := r.URL.Query().Get("name")
	path := filepath.Join(baseDir, filename)
	data, err := os.ReadFile(path)
	if err != nil {
		http.Error(w, "not found", 404)
		return
	}
	w.Write(data)
}

File: skills/vuln-discovery/evals/fixtures/python/path_traversal_vulnerable.py
"""Vulnerable: Path traversal via user-controlled filename."""
import os
from flask import Flask, request, send_file

app = Flask(__name__)
UPLOAD_DIR = "/var/uploads"

@app.route("/download")
def download():
    filename = request.args.get("file")
    filepath = os.path.join(UPLOAD_DIR, filename)
    return send_file(filepath)

@app.route("/read")
def read():
    path = request.args.get("path")
    with open(path) as f:
        return f.read()
